### ISSUE TITLE
Restirct user's access to the data

### ISSUE THREAD

**Issue by vat78** [Skyro-ph/zeebe-simple-monitor#13]
Implement a security feature that restricts access to data based on the security privileges assigned to users. Users should only be able to view information pertaining to processes they are authorized to access. The user privileges, indicating allowed processes, will be stored in the main database.
For example, privileges might be stored in a table with columns:
- user id
- BPMN process id
- type of privileges.
And the user will be able to see or to manage (based on privilege type) all data related to the allowed process.
User identifier should be taken from the JWT token.

**Comment by StaZisS** [comment]
What are the best tools to restrict access to the process? I know that you can do it using PostFilter, but this method works only with collections of items and I don't think that the best solution would be to pull the whole collection and then filter it only. So far I have started to request from the database the id of the processes available to the client, and work on the basis of this data.

**Comment by vat78** [comment]
Yes, I think the simplest solution is to add user id to sql requests and filter data at the DB layer

**Comment by StaZisS** [comment]
What about creating resources? We need to get information from the token that the user has admin rights to create resources. I'm talking about these [enpoints](https://github.com/Skyro-ph/zeebe-simple-monitor/blob/restict-user-access/src/main/java/io/zeebe/monitor/rest/ProcessResource.java).

**Comment by vat78** [comment]
As I understand, those resources are related to the concrete process. And if user has access to this process, he is allowed to create resources for it.
The only question is about uploading new models. If it's impossible to define the process definition key there, lets prohibit this endpoint for now. There are another tools for publishing models.
